app/components/forms/input.tsx

```diff
@@ -103,6 +103,10 @@ const Input = forwardRef(function Input(
     // @TODO dont know how to fix that.
     // @ts-ignore
     input = <textarea {...inputProps} rows={rest.rows || 8} />;
+  } else if (inputType === "number") {
```

Comment:
I dont think you need to do that. The reason we do it in the case of text area is because the node itself has a different name. `<input>` vs `<textarea>`
However for numbers we still use an input and we are using the inputProps so it should directly work, without you having to change anything because default of inputType is `input`.
So you can just use it like:

`<Input type="number" name="whatever" />`